### PR TITLE
Fix LFRic algorithm PSyIR module and subroutine algorithm names error (towards #1618)

### PR DESCRIPTION
**Pull request by rupertford**
The module and subroutine names that are generated by the new algorithm layer PSyIR approach were incorrect (as they used the generic rules).

### CODE REVIEW

**Comment by rupertford** [comment]
Ready for first review from @arporter

File: src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py
Comment (arporter) [inline src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py:396]: This looks overly complicated given that the code below only seems to expect a single value for `result_node` and this test only works with a single input code fragment.

File: src/psyclone/tests/domain/lfric/algorithm/lfric_alg_psyir_test.py
Comment (arporter) [inline src/psyclone/tests/domain/lfric/algorithm/lfric_alg_psyir_test.py:193]: Again, this seems overly complicated?

**Review by arporter — CHANGES_REQUESTED** [review]
This is a small bug fix. No doc. updates required.
Tests and examples are all OK with compilation.
All changes are pylint clean.
Would have gone in first time but the use of a loop within two of the tests puzzles me.

**Comment by arporter** [comment]
I forgot to say - please could you bring up-to-date with master and fix the conflict.

File: src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py
Comment (rupertford) [reply src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py:396]: Good point. I just copied the generic logic without thinking about it. I've now simplified for this particular example.

File: src/psyclone/tests/domain/lfric/algorithm/lfric_alg_psyir_test.py
Comment (rupertford) [reply src/psyclone/tests/domain/lfric/algorithm/lfric_alg_psyir_test.py:193]: Updated as above.

**Comment by rupertford** [comment]
Ready for next review from @arporter

**Comment by arporter** [comment]
I still need to check tests etc. after bringing this branch to master.

**Comment by arporter** [comment]
All tests, examples and docs are still OK after merging master. Just waiting on CI now...

**Pull request merged**
